src/client.ts

```diff
@@ -315,22 +315,29 @@ export class StreamChat<
   _hasConnectionID = () => Boolean(this.connectionID);
 
   /**
-   * setUser - Set the current user, this triggers a connection to the API
+   * connectUser - Set the current user, this triggers a connection to the API
    *
    * @param {UserResponse<UserType>} user Data about this user. IE {name: "john"}
    * @param {TokenOrProvider} userTokenOrProvider Token or provider
    *
    * @return {ConnectAPIResponse<ChannelType, CommandType, UserType>} Returns a promise that resolves when the connection is setup
    */
-  setUser = (
+  connectUser = (
     user: UserResponse<UserType>,
     userTokenOrProvider: TokenOrProvider,
   ): ConnectAPIResponse<ChannelType, CommandType, UserType> => {
     if (this.userID) {
       throw new Error(
-        'Use client.disconnect() before trying to connect as a different user. setUser was called twice.',
+        'Use client.disconnect() before trying to connect as a different user. connectUser or setUser was called twice.',
+      );
+    }
+
+    if (this._isUsingServerAuth() && !this.options.serverConnectUser) {
+      throw new Error(
+        'Do not use connectUser or setUser server side. If you have a valid use-case, add "serverConnectUser: true" to the client options to disable this error.',
```

Comment:
What's the usecase for calling `connectUser` with server auth?

Reply:
We use it in QA, so I guess we can't fully restrict the things we use by ourselves :)

The only use case I can come up with is desktop/CLI client

Reply:
Don't we use `getTestClient(serverSide)` for this?

Reply:
Yeah, you're right, that doesn't make any sense